engine/src/flutter/lib/web_ui/test/common/spy.dart

```diff
@@ -37,16 +50,23 @@ class PlatformMessagesSpy {
 
   bool get _isActive => _callback != null;
 
-  /// List of intercepted messages since the last [setUp] call.
+  /// List of intercepted method calls since the last [setUp] call.
   final List<PlatformMessage> messages = <PlatformMessage>[];
 
+  /// List of intercepted string messages since the last [setUp] call.
+  final List<PlatformStringMessage> strings = <PlatformStringMessage>[];
+
   /// Start spying on platform messages.
   ///
   /// This is typically called inside a test's `setUp` callback.
   void setUp() {
     assert(!_isActive);
     _callback = (String channel, ByteData? data, PlatformMessageResponseCallback? callback) {
-      messages.add(PlatformMessage(channel, const JSONMethodCodec().decodeMethodCall(data)));
+      if (channel == 'flutter/lifecycle') {
+        strings.add(PlatformStringMessage(channel, const StringCodec().decodeMessage(data!)));
```

Comment:
Are we even using the `strings` anywhere in tests? Or is this to be able to listen to all channels, even `flutter/lifecycle`?